### PR TITLE
Introduces generic defillama scrapper

### PR DESCRIPTION
**Pull request by daniel-baez**
# What is this change?

This change introduces directory `apps/shinkai-tool-defillama-tvl-rankings` a simple function taking up to two parameters:

The first parameter is named `categoryName` and it's required, this must be one of the names seen under this [table](https://defillama.com/categories)

The second optional parameter is named `networkName`, and it must be one of: 'Ethereum', 'Solana', 'Arbitrum', 'Base', 'Cardano', 'Near', 'BSC', 'Sui'

Using these two inputs the tool gets the right page from defillama.com returning a csv version of the main table found in it, optionally narrowing down by  network name

### CODE REVIEW

File: libs/shinkai-tools-runner/src/built_in_tools.rs [hunk 1]
```diff
@@ -237,6 +247,16 @@ lazy_static! {
                 .unwrap(),
             )),
         );
+        m.insert(
```
Comment (agallardol) [inline libs/shinkai-tools-runner/src/built_in_tools.rs:250]: Not sure but it seems to be added twice

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@shinkai_protocol/source",
-  "version": "0.7.13",
+  "version": "0.7.12",
```
Comment (agallardol) [inline package.json:3]: Why we are "downgrading" the package version?

**Review by agallardol — CHANGES_REQUESTED** [review]
@daniel-baez is this tool replacing the existing one about defillama? If that's the case maybe we can just remove the old one? :(

**Comment by daniel-baez** [comment]
> @daniel-baez is this tool replacing the existing one about defillama? If that's the case maybe we can just remove the old one? :(

yes, it can make the same results

File: libs/shinkai-tools-runner/src/built_in_tools.rs (hunk 1 above)
Comment (daniel-baez) [reply libs/shinkai-tools-runner/src/built_in_tools.rs:250]: right, let me remove it

File: package.json (hunk 2 above)
Comment (daniel-baez) [reply package.json:3]: m... maybe I pressed `x` by mistake, let me fix

Comment (agallardol) [reply package.json]: image

**Pull request merged**